src/uu/dmesg/src/dmesg.rs

```diff
@@ -217,3 +435,57 @@ impl Record {
         }
     }
 }
+
+impl TryFrom<u32> for Level {
+    type Error = Box<dyn UError>;
+
+    fn try_from(value: u32) -> UResult<Self> {
+        let priority = value & 0b111;
+        match priority {
+            0 => Ok(Level::Emerg),
+            1 => Ok(Level::Alert),
+            2 => Ok(Level::Crit),
+            3 => Ok(Level::Err),
+            4 => Ok(Level::Warn),
+            5 => Ok(Level::Notice),
+            6 => Ok(Level::Info),
+            7 => Ok(Level::Debug),
+            _ => todo!(),
+        }
+    }
+}
+
+impl TryFrom<u32> for Facility {
+    type Error = Box<dyn UError>;
+
+    fn try_from(value: u32) -> Result<Self, Self::Error> {
+        let facility = (value >> 3) as u8;
+        match facility {
+            0 => Ok(Facility::Kern),
+            1 => Ok(Facility::User),
+            2 => Ok(Facility::Mail),
+            3 => Ok(Facility::Daemon),
+            4 => Ok(Facility::Auth),
+            5 => Ok(Facility::Syslog),
+            6 => Ok(Facility::Lpr),
+            7 => Ok(Facility::News),
+            8 => Ok(Facility::Uucp),
+            9 => Ok(Facility::Cron),
+            10 => Ok(Facility::Authpriv),
+            11 => Ok(Facility::Ftp),
+            12 => Ok(Facility::Res0),
+            13 => Ok(Facility::Res1),
+            14 => Ok(Facility::Res2),
+            15 => Ok(Facility::Res3),
+            16 => Ok(Facility::Local0),
+            17 => Ok(Facility::Local1),
+            18 => Ok(Facility::Local2),
+            19 => Ok(Facility::Local3),
+            20 => Ok(Facility::Local4),
+            21 => Ok(Facility::Local5),
+            22 => Ok(Facility::Local6),
+            23 => Ok(Facility::Local7),
+            _ => todo!(),
```

Comment:
See my previous comment.

Reply:
Resolved in same commit as previous comment.